### PR TITLE
initial spec

### PR DESCRIPTION
**Pull request by duglin**
Signed-off-by: Doug Davis <dug@us.ibm.com>
Signed-off-by: Austen Collins <austen@serverless.com>

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+* Description: ID of the event.  Can be specified by the producer.  The
+  semantics of this string are explicitly undefined to ease the implementation
+  of producers.  Enables deduplication.
+* Examples:
+  * A database commit ID
+* Constraints:
+  * Required
+  * Unique per producer
+
+### correlation-id
```
Comment (ac360) [inline spec.md]: @duglin Could we move this to the `Context Attributes Backlog`.  This was suggested in the Google doc by a contributor, but wasn't agreed to by anyone else.  We need to discuss and review new properties before adding them in.

File: spec.md [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+  event. Enables traceability through systems.
+* Examples:
+  * Event A generated with unique correlation-id. Event handler receives event
+    and generates a new event in response, Event B. Event B copies the
+    correlation-id of Event A so that the chain of events can be properly
+    correlated.
+* Constraints:
+  * Optional
+
+### causation-id
```
Comment (ac360) [inline spec.md]: @duglin Could we move this to the `Context Attributes Backlog`.  This was suggested in the Google doc by a contributor, but wasn't agreed to by anyone else.  We need to discuss and review new properties before adding them in.

Comment (duglin) [reply spec.md]: No idea why this comment didn't go away when I moved the entry, but I did move it as you can see.

**Comment by duglin** [comment]
@ac360 can you add a comment saying that you agree with the edit and are "signing off" on it - in essence signing a DCO, just so we transfer right to the group?

**Comment by ac360** [comment]
These edits look good to me.

**Pull request merged**